Mathlib/Analysis/Calculus/TangentCone.lean

```diff
@@ -140,6 +147,14 @@ theorem tangentCone_congr (h : 𝓝[s] x = 𝓝[t] x) : tangentConeAt 𝕜 s x =
 theorem tangentCone_inter_nhds (ht : t ∈ 𝓝 x) : tangentConeAt 𝕜 (s ∩ t) x = tangentConeAt 𝕜 s x :=
   tangentCone_congr (nhdsWithin_restrict' _ ht).symm
 
+end TVS
+
+section Normed
+variable [NormedAddCommGroup E] [NormedSpace 𝕜 E]
+variable [NormedAddCommGroup F] [NormedSpace 𝕜 F]
+variable [NormedAddCommGroup G] [NormedSpace ℝ G]
+variable {x y : E} {s t : Set E}
+
 /-- The tangent cone of a product contains the tangent cone of its left factor. -/
 theorem subset_tangentCone_prod_left {t : Set F} {y : F} (ht : y ∈ closure t) :
```

Comment:
This then becomes the last generalization to resolve in this file.

Reply:
This (the rest of the proof, not what you've written) does not seem obvious to me. I need to think about it. One issue is the hardcoded countability in the definition of the tangent code.

Reply:
Note that I'd be very happy to postpone generalizing this lemma to a later PR; I think the generalization you helped with with that's already in this diff is enough to unlock a lot of basic `deriv` results.